### PR TITLE
Add `typing_extensions.NamedTuple`

### PR DESCRIPTION
**Pull request by AlexWaygood**
Fixes https://github.com/python/typing_extensions/issues/56

### CODE REVIEW

File: stdlib/typing_extensions.pyi [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import abc
+import collections
 import sys
 from _typeshed import IdentityFunction, Self as TypeshedSelf  # see #6932 for why the Self alias cannot have a leading underscore
+from collections.abc import Iterable,
```
Comment (AlexWaygood) [inline stdlib/typing_extensions.pyi:5]:
```suggestion
from collections.abc import Iterable
```

File: stdlib/typing_extensions.pyi [hunk 2]
```diff
@@ -237,3 +241,24 @@ else:
...
+            _field_types: dict[str, type]
+        _field_defaults: dict[str, Any]
+        _fields: tuple[str, ...]
+        _source: str
+        @overload
+        def __init__(self, typename: str, fields: Iterable[tuple[str, Any]] = ...) -> None: ...
+        @overload
+        def __init__(self, typename: str, fields: None = ..., **kwargs: Any) -> None: ...
+        @classmethod
+        def _make(cls: type[_T], iterable: Iterable[Any]) -> _T: ...
```
Comment (AlexWaygood) [inline stdlib/typing_extensions.pyi:258]:
```suggestion
        def _make(cls: type[TypeshedSelf], iterable: Iterable[Any]) -> TypeshedSelf: ...
```

File: stdlib/typing_extensions.pyi [hunk 3]
```diff
@@ -194,8 +197,9 @@ else:
 
 # New things in 3.11
 if sys.version_info >= (3, 11):
-    from typing import (
+    from typing import (  # Ability to create generic NamedTuples is new in 3.11
         LiteralString as LiteralString,
+        NamedTuple as NamedTuple,
```
Comment (AlexWaygood) [inline stdlib/typing_extensions.pyi:202]:
```suggestion
    from typing import (
        LiteralString as LiteralString,
        # Ability to create generic NamedTuples is new in 3.11
        NamedTuple as NamedTuple,
```

**Comment by AlexWaygood** [comment]
> Would anything break if we unconditionally did `from typing import NamedTuple as NamedTuple`? We already do that for `overload` too.

My thinking was that type checkers might want to distinguish `typing_extensions.NamedTuple` from `typing.NamedTuple`, so that they can emit an error if a user tries to create a generic `typing.NamedTuple` on <3.11, since that will fail at runtime. I.e., it would be a similar argument to the argument made in #7130 for declaring `typing_extensions.Protocol` as a distinct symbol to `typing.Protocol`.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**